util/dir/src/lib.rs

```diff
@@ -15,15 +15,35 @@
 // along with OpenEthereum.  If not, see <http://www.gnu.org/licenses/>.
 
 #![warn(missing_docs)]
-
 //! Dir utilities for platform-specific operations
+///
+/// Base ($BASE) Paths where we store our data and cache corresponds to:
+/// Windows:
+///     UserData: FOLDERID_RoamingAppData: %APPDATA% (%USERPROFILE%\AppData\Roaming)
+///     UserCache: FOLDERID_LocalAppData: %LOCALAPPDATA% (%USERPROFILE%\AppData\Local)
+/// MacOS:
+///     UserData: /Users/Alice/Library/Application Support/
+///     UserCache: /Users/Alice/Library/Caches/
+/// Unix:
+///     UserData is: $HOME/.local/share/
+///     UserCache is: $HOME/.cache/
+///
+/// On this UserData base path we are adding additional application folders:
+/// If older parity folders are present we will use them as default for backward compatibility:
+/// Windows: $BASE/Parity/Ethereum/
```

Comment:
I would like to move finally to OE namespace:

Windows: `$BASE/OpenEthereum`
Unix/MaxOS: `$BASE/openethereum`

Reply:
Logic behind this is if there is an old parity path use it, if not create `$BASE/openethereum`.